packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -218,29 +219,44 @@ class DataSourceWithBackend<
   /**
    * Make a GET request to the datasource resource path
    */
-  async getResource(path: string, params?: any): Promise<any> {
-    return getBackendSrv().get(`/api/datasources/${this.id}/resources/${path}`, params);
+  async getResource(path: string, params?: any, options?: Partial<BackendSrvRequest>): Promise<any> {
```

Comment:
if not much to ask, could you type this function to not use `any` (and the following one)

Reply:
No, I think it makes sense 👍 

The only problem is that I think the most I can do is this to use `Record<string, any>` instead of `any` (probably still better), as [the params accept any value](https://github.com/grafana/grafana/blob/607d45436073c843bbb621c0c2a55ed146b5918d/packages/grafana-runtime/src/services/backendSrv.ts#L62) and [get serialised](https://github.com/grafana/grafana/blob/84b7efb393a8320c335a545e2a89f7ab0c8bda4c/public/app/core/utils/fetch.ts#L129-L130) before being added to the URL.

What do you think?

Reply:
Went with `BackendSrvRequest['params'] in the end 👍